### DISCUSSION TITLE
errors and uncertainties on data

### DISCUSSION THREAD

**Discussion by cburgard** [hep-statistics-standard/hep-statistics-standard#27] [Standard]
Hi everyone!

There's a brewing question that we have discussed privately with multiple people (e.g. @oschulz and @Cornelius-G ) a few times in the past, namely on the way uncertainties on data are handled.

Now, one could make the point that data "is what it is" and should never have any uncertainties whatsoever, while the uncertainty should always be part of the model, but this picture breaks down at some point, at the very latest when handling unfolded data, which is actually data that already includes some modelling assumptions that lead to non-trivial correlations of uncertainties.

The current draft of the standard tries (at least a little bit) to capture this issue, by allowing people to assign uncertainties to data.

In the simplest case, this could be a histogram with error bars which would allow a chi2 to be constructed.

```
"data": [
  {
    "name": "data2",
    "type": "binned",
    "contents": [
      9.0,
      18.4
    ],
    "uncertainties": [
      3.0,
      4.5
    ],
    "axes": [
      {
        "name": "variable1",
        "nbins": 2,
        "min": 1,
        "max": 3
      },
      // Add more axis objects if needed
    ]
  },
  // Add more data objects if needed
]
```

In a more complicated case, we would do something like this:

```
"data": [
  {
    "name": "data4",
    "type": "binned",
    "contents": [
      9.0,
      18.4
    ],
    "uncertainty": {
      "type": "gaussian_uncertainty",
      "correlation": 0,
      "sigma": [3, 4]
    },
    "axes": [
      {
        "name": "variable1",
        "nbins": 2,
        "min": 1,
        "max": 3
      },
      // Add more axis objects if needed
    ]
  },
  // Add more data objects if needed
]
```

Similarly, for unbinned data, we currently envision to be able to give error bars per bin

```
"data": [
  {
    "name": "data1",
    "type": "unbinned",
    "weights": [
      9.0,
      18.4
    ],
    "entries": [
      [1, 3],
      [2, 9]
    ],
    "entries_errors": [
      [0.3],
      [0.6]
    ],
    "axes": [
      {
        "name": "variable1",
        "min": 1,
        "max": 3
      },
      {
        "name": "variable2",
        "min": -10,
        "max": 10
      },
      // Add more axis objects if needed
    ]
  },
  // Add more data objects if needed
]
```

Now, here it becomes a bit fishy to decide whether we want to allow both `weights` and `entries_errors` to exist at the same time, and what we want to do in terms of chi2 if both have been defined.

Also, as a side note, we currently (somehwat inconsistenly) use `errors` and `uncertainties` here.

I'd love to hear peoples inputs on whether this approach makes sense, and what your opinions are.

**Comment by jonas-eschle** [comment]
A bit divided on the issue: I understand and acknowledge the need for i.e. histograms; it's impractical not to have it. At the same time, I am not sure how much this is defined: we can assume that the data was produced from a specific distribution (which we simplify often to a Gaussian/Poisson and then give the expected interval as uncertainty with a single value) but that's a shortcut and not very general (especially talking about unfolded dists).

Tendentially, I would go for no uncertainties on the _data_ (we're just talking abouth storing it in the format, not how it's actually represented in the software to the user).

As a more general case, we could have a "distribution" to describe the uncertainty of the data, i.e. instead of some special cases of errors, a general description of how the data was drawn. This can be used to resample a new dataset.
